coderd/database/migrate.go

```diff
@@ -36,8 +46,19 @@ func migrateSetup(db *sql.DB) (source.Driver, *migrate.Migrate, error) {
 }
 
 // MigrateUp runs SQL migrations to ensure the database schema is up-to-date.
-func MigrateUp(db *sql.DB) error {
+func MigrateUp(db *sql.DB) (retErr error) {
 	_, m, err := migrateSetup(db)
+	defer func() {
+		srcErr, dbErr := m.Close()
```

Comment:
Just realized `m` can be nil here which would panic. Should we move this to after `err` is handled?